lib/utils/escape-filter-value.js

```diff
@@ -31,18 +31,17 @@ function escapeBuffer (buf) {
   for (let i = 0; i < buf.length; i += 1) {
     if (buf[i] >= 0xc0 && buf[i] <= 0xdf) {
       // Represents the first byte in a 2-byte UTF-8 character.
-      result += '\\' + buf[i].toString(16) + '\\' + buf[i + 1].toString(16)
+      result += '\\' + buf[i].toString(16);
+      if(i<15) { result += '\\' + buf[i + 1].toString(16); }  //Only add second byte if exist in buffer
       i += 1
       continue
     }
 
     if (buf[i] >= 0xe0 && buf[i] <= 0xef) {
       // Represents the first byte in a 3-byte UTF-8 character.
-      result += [
-        '\\', buf[i].toString(16),
-        '\\', buf[i + 1].toString(16),
-        '\\', buf[i + 2].toString(16)
-      ].join('')
+      result += '\\' + buf[i].toString(16);
+      if(i<15) { result += '\\' + buf[i + 1].toString(16); }  //Only add second byte if exist in buffer
+      if(i<14) { result += '\\' + buf[i + 2].toString(16); }  //Only add third byte if exist in buffer
```

Comment:
Again, where does this `14` constant come from?

Reply:
Same as above. Sorry.
I have commit change to replace this by "buf.length-1" and "buf.length-2".